src/sentry/profiles/task.py

```diff
@@ -136,6 +138,21 @@ def process_profile_task(
         set_measurement("profile.stacks.processed", len(profile["profile"]["stacks"]))
         set_measurement("profile.frames.processed", len(profile["profile"]["frames"]))
 
+    if (
+        enabled := options.get("profiling.generic_metrics.functions_ingestion.enabled")
+    ) and project.organization_id in (
+        allowed_orgs := options.get("profiling.generic_metrics.functions_ingestion.allowed_org_ids")
+    ):
+        try:
+            dsn = get_metrics_dsn(project.id)
```

Comment:
We could instrument this call with a sentry span and/or a `metrics.timer`, to see if it causes a performance bottleneck and to help decide whether we need a larger cache.